repl/src/main/scala/com/cloudera/livy/repl/sparkr/SparkRInterpreter.scala

```diff
@@ -62,19 +67,37 @@ object SparkRInterpreter {
     ).r.unanchored
 
   def apply(): SparkRInterpreter = {
-    val executable = sparkRExecutable
-      .getOrElse(throw new IllegalStateException("Cannot find sparkR executable."))
+    val backendTimeout = sys.env.getOrElse("SPARKR_BACKEND_TIMEOUT", "120").toInt
+    val mirror = universe.runtimeMirror(getClass.getClassLoader)
+    val sparkRBackendClass = mirror.classLoader.loadClass("org.apache.spark.api.r.RBackend")
+    val ctor = sparkRBackendClass.getDeclaredConstructor().newInstance()
+
+    var sparkRBackendPort = 0
+    val initialized = new Semaphore(0)
+    // Launch a SparkR backend server for the R process to connect to
+    val sparkRBackendThread = new Thread("SparkR backend") {
+      override def run() {
+        sparkRBackendPort = sparkRBackendClass.getMethod("init").invoke(ctor).asInstanceOf[Int]
+        initialized.release()
+        sparkRBackendClass.getMethod("run").invoke(ctor)
+      }
+    }
 
-    val builder = new ProcessBuilder(Seq(executable.getAbsolutePath).asJava)
+    sparkRBackendThread.start()
+    // Wait for RBackend initialization to finish
+    initialized.tryAcquire(backendTimeout, TimeUnit.SECONDS)
 
+    val builder = new ProcessBuilder(Seq("R", "--slave @").asJava)
     val env = builder.environment()
+
+    env.put("EXISTING_SPARKR_BACKEND_PORT", sparkRBackendPort.toString)
+    env.put("SPARKR_PACKAGE_DIR", "./sparkr")
+    env.put("R_PROFILE_USER",
+        Seq("./sparkr", "SparkR", "profile", "general.R").mkString(File.separator))
     env.put("SPARK_HOME", sys.env.getOrElse("SPARK_HOME", "."))
-    env.put("SPARKR_DRIVER_R", createFakeShell().toString)
```

Comment:
Can't you just delete the `createFakeShell` method now, along with `fake_R.sh` and `dev/spark/bin/sparkR`?